src/benchmarks/gc/GC.Infrastructure/GC.Infrastructure/Commands/GCPerfSim/GCPerfSimCommand.cs

```diff
@@ -157,6 +157,15 @@ public static GCPerfSimResults RunGCPerfSim(GCPerfSimConfiguration configuration
 
                         foreach (var environVar in environmentVariables)
                         {
+                            // Check if the log file is specified, also store it in a run-specific location.
+                            // This log file should be named in concordance with the name of the run and the benchmark.
+                            if (string.CompareOrdinal(environVar.Key, "DOTNET_GCLogFile") == 0)
+                            {
+                                string gcLogDirectory = Path.Combine(outputPath, runInfo.RunDetails.Key + "_GCLog");
+                                Core.Utilities.TryCreateDirectory(gcLogDirectory);
+                                gcperfsimProcess.StartInfo.WorkingDirectory = gcLogDirectory;
```

Comment:
I don't particularly like this fix.  Currently the infra will just run gcperfsim from the current working directory and leave the gc log there, so this is probably an improvement.  It doesn't handle the server case.

Reply:
I thought this was by design. do you need to be able to specify a different dir? @mrsharm could take a look when he gets back. for server I don't think we ever needed to specify anything other than the current directory (because we just get files in the current dir back).

Reply:
The infra has logic for capturing things like stdout and putting the results in `outputPath`, so it seems inconsistent for the gc log to not end up there.  I did the `_GCLog` thing to match the ASP.NET testing, though with the right filename it probably doesn't need to be in a subdirectory.  The infra is generally creating `outputPath`, so the directory doesn't exist when launching the infra.

The ASP.NET case has some explicit logic asking for the GCLog directory to be retrieved, so I was assuming that something similar was needed (and missing) for gcperfsim.  I haven't actually tried it though.

Reply:
I think a cleaner solution to this would be to update the value of the environment variable of ``DOTNET_GCLogFile`` path to point to the output path combined with the new folder (*_GCLog) and then the name of the file specified with the environment variable; this way we don't need to change anything with the way the process will be run and will have made a decision for the user that's consistent with the ASP.NET functionality.

Reply:
I pushed a change to do this (still need to test it though).

Note that this requires a casing match.  I could enumerate the variables and do case-insensitive matches if you'd prefer.  (I think it would be foreach-to-identify, update-outside-of-foreach, foreach-to-set-env-vars.  Technically I could just do it when setting the envvars, but I didn't like leaving the envvar dictionary in the "wrong" state as it seemed like it could invite a mistake in the future in code later in the method.)

Also, if the variable has an absolute path, then this code won't do anything.  I think I'm ok with that.

Reply:
>> do case-insensitive matches if you'd prefer

No preference. 

>> Also, if the variable has an absolute path, then this code won't do anything. I think I'm ok with that. 

Yes, my rationale here was to main consistency with the output_path that the user has expressed intention where the outputs should live vs. an arbitrary path. I'd rather have everything in one place (enforced by the Infra).